### PR TITLE
Set bodyproducer with empty content for POST

### PR DESCRIPTION
**Pull request by redapple**
Fixes https://github.com/scrapy/scrapy/issues/2677

### CODE REVIEW

File: scrapy/core/downloader/handlers/http11.py [hunk 1]
```diff
@@ -280,19 +280,7 @@ def download_request(self, request):
...
-            #
-            # RFC 7230#section-3.3.2:
-            # "a Content-Length header field is normally sent in a POST
-            # request even when the value is 0 (indicating an empty payload body)."
-            #
-            # Twisted Agent will not add "Content-Length: 0" by itself
-            if method == b'POST':
-                headers.addRawHeader(b'Content-Length', b'0')
-
+            bodyproducer = _RequestBodyProducer(b'') if method == b'POST' else None
```
Comment (dangra) [inline scrapy/core/downloader/handlers/http11.py]: I would go for:
```python
if request.body:
  bodyproducer = _RequestBodyProducer(request.body)
elif method == b'POST':
  bodyproducer = _RequestBodyProducer(b'')
else:
  bodyproducer = None
```

Comment (dangra) [inline scrapy/core/downloader/handlers/http11.py]: btw, what is the expected behavior in terms of Content-Length header for PUT and PATCH requests without body content?

Comment (redapple) [inline scrapy/core/downloader/handlers/http11.py]: Twisted's fix considers both PUT and POST: https://github.com/twisted/twisted/pull/670/

**Comment by dangra** [comment]
looks good except for a missing comment explaining why it needs a special case for POST requests. A comment saying we can drop once minimum twisted version is bumped above 14.0.0

Also, a test case checking content-length is set only once as safe check for past and future compatibility with twisted.

**Comment by redapple** [comment]
@dangra , PR updated.

**Pull request merged**